packages/sdk-js/src/auto-wrapper.ts

```diff
@@ -233,20 +233,43 @@ if (
 const gb = new GrowthBook({
   ...dataContext,
   remoteEval: !!dataContext.remoteEval,
-  trackingCallback: (e, r) => {
-    const p = { experiment_id: e.key, variation_id: r.key };
+  trackingCallback: async (e, r) => {
+    const promises: Promise<unknown>[] = [];
+    const eventParams = { experiment_id: e.key, variation_id: r.key };
 
     // GA4 - gtag
-    window.gtag && window.gtag("event", "experiment_viewed", p);
+    if (window.gtag) {
+      let gtagResolve;
+      const gtagPromise = new Promise((resolve) => {
+        gtagResolve = resolve;
+      });
+      promises.push(gtagPromise);
+      window.gtag("event", "experiment_viewed", {
+        ...eventParams,
+        event_callback: gtagResolve,
+      });
+    }
 
     // GTM - dataLayer
-    window.dataLayer &&
-      window.dataLayer.push({ event: "experiment_viewed", ...p });
+    if (window.dataLayer) {
+      let datalayerResolve;
+      const datalayerPromise = new Promise((resolve) => {
+        datalayerResolve = resolve;
+      });
+      promises.push(datalayerPromise);
+      window.dataLayer.push({
+        event: "experiment_viewed",
+        ...eventParams,
+        eventCallback: datalayerResolve,
+      });
+    }
 
     // Segment - analytics.js
-    window.analytics &&
-      window.analytics.track &&
-      window.analytics.track("Experiment Viewed", p);
+    if (window.analytics && window.analytics.track) {
+      window.analytics.track("Experiment Viewed", eventParams);
```

Comment:
If someone is only using Segment, previously, this trackingCallback would return void, which would cause it to use the `navigateDelay`.  Now, it will return `Promise<void>` which resolves immediately, which will cause it to immediately redirect before Segment could fire.

We should use a promise here instead that waits a hardcoded amount of time.  Segment's docs recommend using a value of `300ms` to give the tracking call enough time to finish.

Reply:
The SDK will always wait for at least the MAX of:
- `context.navigateDelay ?? 100`
- the tracking promise

So we could just set navigateDelay to 300

Reply:
I'd prefer having `navigateDelay` set to something smaller like 100, and have this Segment call specifically wait 300ms.  So they only pay that extra penalty if they are using Segment.  Otherwise, if GA finishes sooner, it doesn't have to wait.